shared/tasks/assets.mjs

```diff
@@ -14,15 +14,27 @@ export async function write (filePath, result) {
   const writeTasks = []
 
   // Files to write
+  /** @type {SourceMap | undefined} */
+  const map = result.map ? JSON.parse(result.map.toString()) : undefined
```

Comment:
We need `map.sources` a few lines down so parse to [`SourceMap`](https://github.com/source-map/source-map-spec) here

Otherwise across Terser, Rollup and PostCSS we see source maps as:

1. JavaScript `object` types that need `JSON.stringify()`
2. Class instances with `.toJSON()` and `.toString()` methods
3. Preprepared `string` type
4. Missing `undefined` type

This can be seen via the **AssetOutput** `@typedef` in this file